### ISSUE TITLE
only part of filename shown

### ISSUE THREAD

**Issue by shinnlu** [alexusmai/laravel-file-manager#74]
i have many chinese filename in the folder, but only show part of filename,
it's seems filename broken with some character like '-' or something i don't know.
please tell me how to fix the problem.

釋迦牟尼佛介紹-2019.jpg => -2019.jpg
變身金母介紹-2019-修改.jpg => -2019-修改.jpg
講授道果-logo.png => -logo.png
穢跡金剛logo-小檔.jpg => logo-小檔.jpg
諸尊名諱20171211更新.jpg => 20171211更新.jpg

![image](https://user-images.githubusercontent.com/570246/65769632-98eb9300-e166-11e9-86db-faedc9dfdb17.png)

**Comment by shinnlu** [comment]
i guess it's OS dependency problem, the issue was on Ubuntu, but no problem on Macbook with same filename.

**Comment by alexusmai** [comment]
Hi.
I checked these file names.. it's work fine for me. (Homestead VM - Ubuntu OS) It may be some server settings issue.

![015](https://user-images.githubusercontent.com/7473505/65786832-355e6700-e160-11e9-97a9-6b30e2ea85d5.png)

**Comment by shinnlu** [comment]
yes, it's so weird, i still don't know how to find the problem.

**Comment by shinnlu** [comment]
i have a clue, please see left side of picture

![image](https://user-images.githubusercontent.com/570246/65835108-d3932e00-e314-11e9-81e4-d830ccd228d9.png)

the basename and path was correct, but filename was wrong, and the file-manager.js show filename with 'filename' which is wrong path.

**Comment by alexusmai** [comment]
To get information about files and folders, I use the standard function - [List Contents](https://flysystem.thephpleague.com/docs/usage/filesystem-api/)

```
Storage::disk($disk)->listContents($path);
```

**Comment by shinnlu** [comment]
you are right, there are something wrong with standard function ListContents().
i have modify the file-manager.js to show basename until find the solution.
![image](https://user-images.githubusercontent.com/570246/65836684-57084b80-e324-11e9-8242-a532d225036b.png)

**Comment by alexusmai** [comment]
You are the first with this problem, and I can't to repeat this problem (using your file names) - I think you'll need to check your server settings (for dev - try laravel homestead - there is no problem with names), maybe you have some issue with 'locales'...

**Comment by shinnlu** [comment]
i have change config/app.php
locale from 'utf8 to 'zh_TW.utf8'
but still not working.

i found the same issue on the native package of Storage, but i still don't know what i can do.
[https://github.com/thephpleague/flysystem/issues/235](url)

**Comment by shinnlu** [comment]
i have found a solution.
[https://wyde.github.io/2017/11/07/How-to-Setup-Locale-in-Ubuntu-16-04/](url)
the result must generate /etc/default/locale and content will be 
```
#  File generated by update-locale
LANG="en_US.UTF-8"
LC_ALL=zh_TW.UTF-8
```
don't forget to restart apache.

**Comment by shinnlu** [comment]
the locale setting does solve the issue, but for some reason i don't know, problem appear randomly.
just fix it by 
`sudo service php7.3-fpm restart`

i will compare phpinfo() for difference next time.

**Comment by shinnlu** [comment]
so weird, nothing different between phpinfo
[compare two phpinfo](https://github.com/shinnlu/php_setlocale/commit/08b13ee640df0d16c93119295c0017ee9e1ba8b4#diff-60960185de565a2911916df1268671c8)
